server/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -253,7 +253,7 @@ private void dispatchRequest(RestRequest request, RestChannel channel, RestHandl
             if (handler.allowsUnsafeBuffers() == false) {
                 request.ensureSafeBuffers();
             }
-
+            //TODO why we are using client.threadPool().getThreadContext() ?
```

Comment:
I wonder if we should not be using the thread context passed over via `AbstractHttpServerTransport#dispatchRequest`
The thread pool is first created in node and is the used in both `RestController` and `AbstracthttpServerTransport`

Reply:
OOF! Good catch. This is a mistake IMO; we should do one of the following things:

1. Remove ThreadContext from the request dispatcher API and always use the one from the node client
2. Leave the API alone and pass the ThreadContext from the dispatcher to all the other methods
3. Add an assertion in dispatch*Request methods that assert the passed threadcontext is the same as that on the node client

This mainly is just an issue for testing but it is tricky and would be nice if we just had one way to access the value